### PR TITLE
Basic authentication sample using Buddy Auth.

### PR DESCRIPTION
**Pull request by ddeaguiar**
A simple example demonstrating integration with the Buddy Auth auth/authz library. Pertains to issue #500.

### CODE REVIEW

File: samples/buddy-auth/src/buddy_auth/service.clj [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+           :role         :user}
+   :gmw   {:display-name "Gerald M. Weinberg"
+           :password     "rutabaga"
+           :role         :admin}})
+
+(def basic-auth-backend
+  "A buddy-auth Basic Authentication backend.  See
+  https://funcool.github.io/buddy-auth/latest/#http-basic"
+  (auth.backends/basic {:realm  "MyApi"
+                        :authfn (fn [_ authdata]
```
Comment (ohpauleez) [inline samples/buddy-auth/src/buddy_auth/service.clj]: Could you replace the unused arg `_` with something named, just so new users have some context what that arg is for?  Thanks

File: samples/buddy-auth/src/buddy_auth/service.clj [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  'unauthorized' exception."
+  [request]
+  (let [known-user (get users (:identity request))]
+    (if (= :admin (:role known-user))
+      (ring-resp/response  "Only admins can see this!")
+      (buddy.auth/throw-unauthorized))))
+
+(def authentication-interceptor
+  "Port of buddy-auth's wrap-authentication middleware."
+  (on-request ::authenticate auth.middleware/authentication-request basic-auth-backend))
```
Comment (ohpauleez) [inline samples/buddy-auth/src/buddy_auth/service.clj]: Avoid using the helper and make this a direct call the the `interceptor` API function using a map.  Thanks!

File: samples/buddy-auth/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+`wrap-authorization` middleware functions to interceptors.
+
+## Getting Started
+
+1. Start the application: `lein run`
+2. Go to [localhost:8080](http://localhost:8080/) to see: `Hello
+   anonymous`.
+3. Make a request to [localhost:8080](http://localhost:8080/) using
+   HTTP Basic authentication with the username `aaron` and password
+   `secret` to see: `Hello Aaron Aardvark`.
```
Comment (ohpauleez) [inline samples/buddy-auth/README.md]: Could you add a short sentence here to the effect of, "To make an authorization request, set the header..."

File: samples/buddy-auth/src/buddy_auth/service.clj (hunk 1 above)
Comment (ddeaguiar) [inline samples/buddy-auth/src/buddy_auth/service.clj]: Will do.

File: samples/buddy-auth/src/buddy_auth/service.clj (hunk 2 above)
Comment (ddeaguiar) [inline samples/buddy-auth/src/buddy_auth/service.clj]: Ok

File: samples/buddy-auth/README.md (hunk 3 above)
Comment (ddeaguiar) [inline samples/buddy-auth/README.md]: I sure can.

**Comment by ohpauleez** [comment]
Thanks @ddeaguiar!  It's really great to see this coming together.

**Comment by ddeaguiar** [comment]
@ohpauleez changes made. Thanks for the feedback :)

**Comment by ddeaguiar** [comment]
@ohpauleez, is there anything else you think needs addressing before this PR can be merged?

**Comment by ohpauleez** [comment]
@ddeaguiar Let me make one more pass on this next week and we'll get this landed.  Thanks so much for pulling it all together.

**Comment by ohpauleez** [comment]
All this looks fine to me.  Sorry to leave this one sitting so long!

**Pull request merged**